Adopt smart pointers in WebProcessProxy

cdumez opened the pull request:
#### 6c1701ab4621e2fbf852e03374a60e8153163a86
<pre>
Adopt smart pointers in WebProcessProxy
<a href="https://bugs.webkit.org/show_bug.cgi?id=263820">https://bugs.webkit.org/show_bug.cgi?id=263820</a>

Reviewed by Ryosuke Niwa.

* Source/WebKit/Shared/WebBackForwardListItem.h:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageCopyRelatedPages):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::getBrowsingContexts):
* Source/WebKit/UIProcess/HighPerformanceGraphicsUsageSampler.cpp:
(WebKit::HighPerformanceGraphicsUsageSampler::timerFired):
* Source/WebKit/UIProcess/PerActivityStateCPUUsageSampler.cpp:
(WebKit::PerActivityStateCPUUsageSampler::pageForLogging const):
* Source/WebKit/UIProcess/ProvisionalPageProxy.h:
* Source/WebKit/UIProcess/RemotePageProxy.cpp:
(WebKit::RemotePageProxy::protectedPage const):
* Source/WebKit/UIProcess/RemotePageProxy.h:
* Source/WebKit/UIProcess/WebBackForwardCache.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::~WebPageProxy):
(WebKit::WebPageProxy::nonEphemeralWebPageProxy):
(WebKit::WebPageProxy::protectedVisitedLinkStore):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::anyProcessPoolNeedsUIBackgroundAssertion):
(WebKit::WebProcessPool::forEachProcessForSession):
(WebKit::WebProcessPool::checkedWebProcessCache):
(WebKit::WebProcessPool::checkedBackForwardCache):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::allProcesses):
(WebKit::WebProcessProxy::processForIdentifier):
(WebKit::WebProcessProxy::globalPages):
(WebKit::WebProcessProxy::pages const):
(WebKit::WebProcessProxy::forWebPagesWithOrigin):
(WebKit::WebProcessProxy::allowedFirstPartiesForCookies):
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::m_routingArbitrator):
(WebKit::WebProcessProxy::~WebProcessProxy):
(WebKit::WebProcessProxy::updateRegistrationWithDataStore):
(WebKit::WebProcessProxy::addProvisionalPageProxy):
(WebKit::WebProcessProxy::addRemotePageProxy):
(WebKit::WebProcessProxy::getLaunchOptions):
(WebKit::WebProcessProxy::shouldSendPendingMessage):
(WebKit::WebProcessProxy::processWillShutDown):
(WebKit::WebProcessProxy::startDisplayLink):
(WebKit::WebProcessProxy::stopDisplayLink):
(WebKit::WebProcessProxy::setDisplayLinkPreferredFramesPerSecond):
(WebKit::WebProcessProxy::setDisplayLinkForDisplayWantsFullSpeedUpdates):
(WebKit::WebProcessProxy::shutDown):
(WebKit::WebProcessProxy::webPage):
(WebKit::WebProcessProxy::audioCapturingWebPage):
(WebKit::WebProcessProxy::webPageWithActiveXRSession):
(WebKit::WebProcessProxy::notifyPageStatisticsAndDataRecordsProcessed):
(WebKit::WebProcessProxy::notifyWebsiteDataScanForRegistrableDomainsFinished):
(WebKit::WebProcessProxy::notifyWebsiteDataDeletionForRegistrableDomainsFinished):
(WebKit::WebProcessProxy::createWebPage):
(WebKit::WebProcessProxy::addExistingWebPage):
(WebKit::WebProcessProxy::removeWebPage):
(WebKit::WebProcessProxy::updateBackForwardItem):
(WebKit::WebProcessProxy::getNetworkProcessConnection):
(WebKit::WebProcessProxy::createGPUProcessConnection):
(WebKit::WebProcessProxy::gpuProcessDidFinishLaunching):
(WebKit::WebProcessProxy::gpuProcessExited):
(WebKit::WebProcessProxy::didReceiveMessage):
(WebKit::WebProcessProxy::didReceiveSyncMessage):
(WebKit::WebProcessProxy::processDidTerminateOrFailedToLaunch):
(WebKit::WebProcessProxy::didBecomeUnresponsive):
(WebKit::WebProcessProxy::didBecomeResponsive):
(WebKit::WebProcessProxy::willChangeIsResponsive):
(WebKit::WebProcessProxy::didChangeIsResponsive):
(WebKit::WebProcessProxy::setIgnoreInvalidMessageForTesting):
(WebKit::WebProcessProxy::didFinishLaunching):
(WebKit::WebProcessProxy::didDestroyFrame):
(WebKit::WebProcessProxy::recordUserGestureAuthorizationToken):
(WebKit::WebProcessProxy::postMessageToRemote):
(WebKit::WebProcessProxy::closeRemoteFrame):
(WebKit::WebProcessProxy::renderTreeAsText):
(WebKit::WebProcessProxy::maybeShutDown):
(WebKit::WebProcessProxy::canTerminateAuxiliaryProcess):
(WebKit::WebProcessProxy::windowServerConnectionStateChanged):
(WebKit::WebProcessProxy::updateAudibleMediaAssertions):
(WebKit::WebProcessProxy::updateMediaStreamingActivity):
(WebKit::WebProcessProxy::isResponsive):
(WebKit::WebProcessProxy::isResponsiveWithLazyStop):
(WebKit::WebProcessProxy::logDiagnosticMessageForResourceLimitTermination):
(WebKit::WebProcessProxy::didExceedCPULimit):
(WebKit::WebProcessProxy::updateBlobRegistryPartitioningState const):
(WebKit::WebProcessProxy::didCollectPrewarmInformation):
(WebKit::WebProcessProxy::didStartProvisionalLoadForMainFrame):
(WebKit::WebProcessProxy::reportProcessDisassociatedWithPageIfNecessary):
(WebKit::WebProcessProxy::createSpeechRecognitionServer):
(WebKit::WebProcessProxy::muteCaptureInPagesExcept):
(WebKit::WebProcessProxy::disableRemoteWorkers):
(WebKit::WebProcessProxy::enableRemoteWorkers):
(WebKit::WebProcessProxy::protectedWebsiteDataStore const):
(WebKit::WebProcessProxy::getNotifications):
(WebKit::WebProcessProxy::setAppBadge):
(WebKit::WebProcessProxy::setClientBadge):
(WebKit::WebProcessProxy::logger):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::isBlobRegistryPartitioningEnabled const):

Canonical link: <a href="https://commits.webkit.org/269895@main">https://commits.webkit.org/269895@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/668ea96d7cd2fdb55cc5f7e9a19e1ae7e02b30ff

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23994 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2106 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25082 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26135 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22100 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3726 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24477 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22589 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24238 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1638 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20735 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26724 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1394 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27881 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21871 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21935 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25674 "Found 58 new API test failures: /WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/ephemeral, /WebKitGTK/TestLoaderClient:/webkit/WebKitWebView/reload, /WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/cookies, /WebKitGTK/TestSSL:/webkit/WebKitWebView/web-socket-tls-errors, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/usermedia-enumeratedevices-permission-check, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/document/load-events, /WebKitGTK/TestCookieManager:/webkit/WebKitCookieManager/delete-cookies, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/file-chooser-request, /WebKitGTK/TestAuthentication:/webkit/Authentication/authentication-cancel, /WebKitGTK/TestContextMenu:/webkit/WebKitWebView/populate-menu ... (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1331 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19009 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1350 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1742 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3059 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1682 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -183,46 +183,44 @@ auto WebProcessProxy::globalPageMap() -> WebPageProxyMap&
-        if (page)
-            result.append(std::make_pair(page->process().coreProcessIdentifier(), RegistrableDomain(URL(page->currentURL()))));
-    }
+    for (auto& page : globalPages())
```
See CheckedRef?

cdumez replied:
auto& is already a `Ref&` here. globalPages() returns a `Vector<Ref<WebPageProxy>>`

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -639,30 +642,26 @@ void WebProcessProxy::shutDown()
-        if (page)
-            page->disconnectFramesFromPage();
-    }
+    for (auto& page : pages())
```
Use CheckedRef/Ref?

cdumez replied:
pages() returns a `Vector<Ref<WebPageProxy>>`. I guess I can use `Ref&&` instead of `auto&`.

cdumez replied:
`Ref&` or `Ref&&` doesn't build. I'll just use `Ref` then. It's a bit more ref counting churn than `auto&` though.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -674,19 +673,19 @@ void WebProcessProxy::shutDown()
 
 RefPtr<WebPageProxy> WebProcessProxy::audioCapturingWebPage()
 {
     for (auto& page : globalPages()) {
```
Use CheckedRef/Ref?

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -705,26 +704,20 @@ RefPtr<WebPageProxy> WebProcessProxy::webPageWithActiveXRSession()
-        if (page)
-            page->postMessageToInjectedBundle("WebsiteDataScanForRegistrableDomainsFinished"_s, nullptr);
-    }
+    for (auto& page : globalPages())
```
Use CheckedRef/Ref?

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -705,26 +704,20 @@ RefPtr<WebPageProxy> WebProcessProxy::webPageWithActiveXRSession()
-        if (page)
-            page->postMessageToInjectedBundle("WebsiteDataScanForRegistrableDomainsFinished"_s, nullptr);
-    }
+    for (auto& page : globalPages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -705,26 +704,20 @@ RefPtr<WebPageProxy> WebProcessProxy::webPageWithActiveXRSession()
-        if (page)
-            page->postMessageToInjectedBundle("WebsiteDataDeletionForRegistrableDomainsFinished"_s, nullptr);
-    }
+    for (auto& page : globalPages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1031,25 +1024,21 @@ void WebProcessProxy::createGPUProcessConnection(IPC::Connection::Handle&& conne
-        if (page)
-            page->gpuProcessDidFinishLaunching();
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1031,25 +1024,21 @@ void WebProcessProxy::createGPUProcessConnection(IPC::Connection::Handle&& conne
-        if (page)
-            page->gpuProcessExited(reason);
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1163,10 +1150,8 @@ void WebProcessProxy::processDidTerminateOrFailedToLaunch(ProcessTerminationReas
-        if (page)
-            page->dispatchProcessDidTerminate(reason);
-    }
+    for (auto& page : pages)
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1198,10 +1183,8 @@ void WebProcessProxy::didBecomeUnresponsive()
-        if (page)
-            page->processDidBecomeUnresponsive();
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1220,33 +1203,27 @@ void WebProcessProxy::didBecomeResponsive()
-        if (page)
-            page->processDidBecomeResponsive();
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1220,33 +1203,27 @@ void WebProcessProxy::didBecomeResponsive()
-        if (page)
-            page->willChangeProcessIsResponsive();
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1220,33 +1203,27 @@ void WebProcessProxy::didBecomeResponsive()
-        if (page)
-            page->didChangeProcessIsResponsive();
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1299,8 +1277,8 @@ void WebProcessProxy::didFinishLaunching(ProcessLauncher* launcher, IPC::Connect
 #if PLATFORM(MAC)
-    for (const auto& page : pages()) {
-        if (page && page->preferences().backgroundWebContentRunningBoardThrottlingEnabled())
+    for (auto& page : pages()) {
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1530,10 +1502,8 @@ void WebProcessProxy::updateTextCheckerState()
-        if (page)
-            page->activityStateDidChange(ActivityState::IsVisuallyIdle);
-    }
+    for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1776,10 +1746,8 @@ void WebProcessProxy::didChangeThrottleState(ProcessThrottleState type)
-            if (page)
-                page->processWillBecomeSuspended();
-        }
+        for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -1794,10 +1762,8 @@ void WebProcessProxy::didChangeThrottleState(ProcessThrottleState type)
-            if (page)
-                page->processWillBecomeForeground();
-        }
+        for (auto& page : pages())
```
Ditto.

rniwa commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2152,8 +2120,8 @@ PAL::SessionID WebProcessProxy::sessionID() const
     RefPtr<WebPageProxy> targetPage;
-    for (auto& page : pages()) {
-        if (page && page->webPageID() == identifier) {
+    for (auto&& page : pages()) {
```
Why &&? But also use Ref/CheckedRef instead?

cdumez committed:
Adopt smart pointers in WebProcessProxy
https://bugs.webkit.org/show_bug.cgi?id=263820

Reviewed by Ryosuke Niwa.

* Source/WebKit/Shared/WebBackForwardListItem.h:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageCopyRelatedPages):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::getBrowsingContexts):
* Source/WebKit/UIProcess/HighPerformanceGraphicsUsageSampler.cpp:
(WebKit::HighPerformanceGraphicsUsageSampler::timerFired):
* Source/WebKit/UIProcess/PerActivityStateCPUUsageSampler.cpp:
(WebKit::PerActivityStateCPUUsageSampler::pageForLogging const):
* Source/WebKit/UIProcess/ProvisionalPageProxy.h:
* Source/WebKit/UIProcess/RemotePageProxy.cpp:
(WebKit::RemotePageProxy::protectedPage const):
* Source/WebKit/UIProcess/RemotePageProxy.h:
* Source/WebKit/UIProcess/WebBackForwardCache.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::~WebPageProxy):
(WebKit::WebPageProxy::nonEphemeralWebPageProxy):
(WebKit::WebPageProxy::protectedVisitedLinkStore):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::anyProcessPoolNeedsUIBackgroundAssertion):
(WebKit::WebProcessPool::forEachProcessForSession):
(WebKit::WebProcessPool::checkedWebProcessCache):
(WebKit::WebProcessPool::checkedBackForwardCache):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::allProcesses):
(WebKit::WebProcessProxy::processForIdentifier):
(WebKit::WebProcessProxy::globalPages):
(WebKit::WebProcessProxy::pages const):
(WebKit::WebProcessProxy::forWebPagesWithOrigin):
(WebKit::WebProcessProxy::allowedFirstPartiesForCookies):
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::m_routingArbitrator):
(WebKit::WebProcessProxy::~WebProcessProxy):
(WebKit::WebProcessProxy::updateRegistrationWithDataStore):
(WebKit::WebProcessProxy::addProvisionalPageProxy):
(WebKit::WebProcessProxy::addRemotePageProxy):
(WebKit::WebProcessProxy::getLaunchOptions):
(WebKit::WebProcessProxy::shouldSendPendingMessage):
(WebKit::WebProcessProxy::processWillShutDown):
(WebKit::WebProcessProxy::startDisplayLink):
(WebKit::WebProcessProxy::stopDisplayLink):
(WebKit::WebProcessProxy::setDisplayLinkPreferredFramesPerSecond):
(WebKit::WebProcessProxy::setDisplayLinkForDisplayWantsFullSpeedUpdates):
(WebKit::WebProcessProxy::shutDown):
(WebKit::WebProcessProxy::webPage):
(WebKit::WebProcessProxy::audioCapturingWebPage):
(WebKit::WebProcessProxy::webPageWithActiveXRSession):
(WebKit::WebProcessProxy::notifyPageStatisticsAndDataRecordsProcessed):
(WebKit::WebProcessProxy::notifyWebsiteDataScanForRegistrableDomainsFinished):
(WebKit::WebProcessProxy::notifyWebsiteDataDeletionForRegistrableDomainsFinished):
(WebKit::WebProcessProxy::createWebPage):
(WebKit::WebProcessProxy::addExistingWebPage):
(WebKit::WebProcessProxy::removeWebPage):
(WebKit::WebProcessProxy::updateBackForwardItem):
(WebKit::WebProcessProxy::getNetworkProcessConnection):
(WebKit::WebProcessProxy::createGPUProcessConnection):
(WebKit::WebProcessProxy::gpuProcessDidFinishLaunching):
(WebKit::WebProcessProxy::gpuProcessExited):
(WebKit::WebProcessProxy::didReceiveMessage):
(WebKit::WebProcessProxy::didReceiveSyncMessage):
(WebKit::WebProcessProxy::processDidTerminateOrFailedToLaunch):
(WebKit::WebProcessProxy::didBecomeUnresponsive):
(WebKit::WebProcessProxy::didBecomeResponsive):
(WebKit::WebProcessProxy::willChangeIsResponsive):
(WebKit::WebProcessProxy::didChangeIsResponsive):
(WebKit::WebProcessProxy::setIgnoreInvalidMessageForTesting):
(WebKit::WebProcessProxy::didFinishLaunching):
(WebKit::WebProcessProxy::didDestroyFrame):
(WebKit::WebProcessProxy::recordUserGestureAuthorizationToken):
(WebKit::WebProcessProxy::postMessageToRemote):
(WebKit::WebProcessProxy::closeRemoteFrame):
(WebKit::WebProcessProxy::renderTreeAsText):
(WebKit::WebProcessProxy::maybeShutDown):
(WebKit::WebProcessProxy::canTerminateAuxiliaryProcess):
(WebKit::WebProcessProxy::windowServerConnectionStateChanged):
(WebKit::WebProcessProxy::updateAudibleMediaAssertions):
(WebKit::WebProcessProxy::updateMediaStreamingActivity):
(WebKit::WebProcessProxy::isResponsive):
(WebKit::WebProcessProxy::isResponsiveWithLazyStop):
(WebKit::WebProcessProxy::logDiagnosticMessageForResourceLimitTermination):
(WebKit::WebProcessProxy::didExceedCPULimit):
(WebKit::WebProcessProxy::updateBlobRegistryPartitioningState const):
(WebKit::WebProcessProxy::didCollectPrewarmInformation):
(WebKit::WebProcessProxy::didStartProvisionalLoadForMainFrame):
(WebKit::WebProcessProxy::reportProcessDisassociatedWithPageIfNecessary):
(WebKit::WebProcessProxy::createSpeechRecognitionServer):
(WebKit::WebProcessProxy::muteCaptureInPagesExcept):
(WebKit::WebProcessProxy::disableRemoteWorkers):
(WebKit::WebProcessProxy::enableRemoteWorkers):
(WebKit::WebProcessProxy::protectedWebsiteDataStore const):
(WebKit::WebProcessProxy::getNotifications):
(WebKit::WebProcessProxy::setAppBadge):
(WebKit::WebProcessProxy::setClientBadge):
(WebKit::WebProcessProxy::logger):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::isBlobRegistryPartitioningEnabled const):

Canonical link: https://commits.webkit.org/269895@main
